### DISCUSSION TITLE
user-guide/sync

### DISCUSSION THREAD

**Discussion by giscus** [zotero-chinese/wiki#80] [文档评论区]
# user-guide/sync

我们在这里把同步分为“数据的同步”和“文件的同步”。 数据的同步指 条目字段信息、`批注、笔记 的同步，这些数据仅能通过 Zotero 官方提供的服务同步。 文件的同步指 PDF 附件、网页附件、Word 文档、EPub 文件 等所有附件的同步。这些文件既可以通过 Zotero 官方服务同步，也可以通过 WebDAV 网盘来同步。 无论你选择何种同步方...

https://zotero-chinese.com/user-guide/sync.html

**Comment by xxtlalala** [comment]
请问ios端同步设置成功后，笔记里的图片不显示是哪出了问题

**Reply by northword** [reply]
需要确认附件是否设置了同步，如果用官方的同步服务，空间是否足够？如果是webdav，是否两端都已正确设置？pc端右上角绿色小圆圈再点一下同步一次，看有无报错？

**Comment by Cooper1023** [comment]
催更第三方网盘同步附件，更新了希望能踢踢我

**Comment by Ashley9858** [comment]
求问已下载 ZotFile插件，但是条目信息、批注、笔记并没有实现同步（或者说并不知道同步到哪里去了），是因为用WebDAV同步附件造成的吗？？？

**Reply by Cooper1023** [reply]
zotfile只移动了附件pdf的位置吧（manage attachments——rename and move），条目信息、批注、笔记应该都是跟着zotero自己的云端走的。（我是按照网上使用坚果云和zotfile同步的教程设置的）

**Reply by northword** [reply]
文档开头有些：

> 我们在这里把同步分为“数据的同步”和“文件的同步”。
>
> 数据的同步指 条目字段信息、批注、笔记 的同步，这些数据仅能通过 Zotero 官方提供的服务同步。
>
> 文件的同步指 PDF 附件、网页附件、Word 文档、EPub 文件 等所有附件的同步。这些文件既可以通过 Zotero 官方服务同步，也可以通过 WebDAV 网盘来同步。

![image](https://github.com/zotero-chinese/.github/assets/44738481/bb6313ae-0767-499c-8f51-cdee013ea5a8)

**Reply by northword** [reply]
此外，zotfile 与 webdav 在多数情况下并不兼容，建议使用webdav，不建议使用 zotfile。

> 我们不建议使用第三方网盘结合 ZotFile 的方式进行同步。这会导致很多问题。除非你具备较强的问题排查能力，且对 ZotFile 所提供的功能有刚性需求，否则请不要使用这种方法！

**Comment by kidcando** [comment]
我用的是alist+百度云盘进行备份，算是曲线救国

**Reply by killer48** [reply]
大佬求教，请问如何通过alist+百度云盘进行备份，我通过百度云同步盘同步了两个win系统端，但在同步ipad的时候发现百度网盘不支持webdev呜呜

**Reply by kidcando** [reply]
用alist挂百度云，记住路径。配置好密码就可以了。注意一点这里面用的是http  不是https
![image](https://github.com/zotero-chinese/wiki/assets/35875612/8dd50fb9-a47f-4f17-8826-c05ffd4101e5)

**Reply by killer48** [reply]
好嘞，我明天尝试挂载，多谢深夜解惑！

**Reply by FeiLZX3Km** [reply]
大佬，你这样同步可以吗？我设置同步时，zotero提示不是有效的webdav url地址。

**Reply by FeiLZX3Km** [reply]
找到问题了，路径上的一些字符填反了。

**Comment by txctxc** [comment]
请问syncthing 可以同步zotero吗？会不会导致很多问题？

**Reply by kidcando** [reply]
可以同步的，Zotero用的是webdev,只要支持这个都可以同步。

**Reply by jacks935** [reply]
syncthing安卓端应该没有webdev的功能吧，能不能用syncthing同步附件的本地文件夹呢，官方不愿意加这种同步方式吗

**Reply by kidcando** [reply]
不建议这样做，可能会出错，我之前试过用百度云同步文件夹的方式，会出错

**Reply by northword** [reply]
> syncthing安卓端应该没有webdev的功能吧，

只要syncthing提供了WebDAV，就可用，只需要在zotero桌面端和安卓端分别配置WebDAV信息即可。
syncthing安卓端支持WebDAV与否不影响，因为webdav是直接从服务器请求文件，而不是靠网盘官方的客户端下载文件。

> 能不能用syncthing同步附件的本地文件夹呢，官方不愿意加这种同步方式吗

官方不会支持这种方式，因为这是 zotero 不可控的。
你可以仅同步附件文件夹，注意，仅同步附件文件夹，不要同步数据库，不要同步数据库本身！
此外，我们不推荐这种方式，WebDAV可用时，他的可用性远高于使用第三方网盘同步文件夹的方式。

**Reply by jacks935** [reply]
syncthing采用的是去中心化的P2P的同步方式，要在安卓端或Windows端分别选择具体的文件夹，请问安卓附件文件夹路径一般在哪呀

**Reply by jacks935** [reply]
因为最近刚出的zotero beta还没有webdev的同步方式，就想试试syncthing

**Reply by jacks935** [reply]
/data/data/org.zotero.android/files/downloads/custom_my_library/
找到了，但是在根目录，要root权限，syncthing也访问不了，那就只能用webdev了

**Comment by 123yiyiyi** [comment]
请问我在zotero客户端中设置了infinicloud同步 验证显示成功，但infinicloud官网zotero文件还是空的，只有在zotero客户端重置在线图书馆后才有数据这是为什么呢 点击右侧同步箭头没有用

**Reply by guaguastandup** [reply]
重置在线图书馆是哪个操作呀？
我从坚果云迁移到infinicloud，然后用attcher把所有pdf重新命名了，之后就正常同步了

**Comment by mmo262** [comment]
zotfile + WPS没出过bug

**Comment by Charagol** [comment]
**仅代表个人观点，可能有错误，你可以尝试但我不对数据库负责！**
---
不建议直接同步C:\Users\[Username]\Zotero数据库的原因是，

1. 首先知道笔记同步技术是SQlite，我理解的是，Zotero对文档做所的笔记（高亮或文字等）什么的，都是储存在zotero.sqlite-journal这个临时文件中，相当于增量文件，当你关闭zotero时该文件会合并到zotero.sqlite中然后消失。所以是记录在zotero.sqlite中，
2. 而后，Zotero在实时云同步软件中会出错的原因就是zotero.sqlite-journal锁定。Zotero打开时会创建并锁定这个临时文件以便更改，与此同时云同步的软件也会检测到这个文件并锁定上传，导致这个文件被同时锁定，进而出错，结果就是在Zotero关闭后文件还在（由网盘试图上传）。
3. 结果的结果就是Zotero在下次打开准备创建zotero.sqlite-journal文件时，上一次的还在，直接报错不让你进去，有时候网盘聪明点会给你留个zotero.sqlite的冲突文件。

我没弄官方云和Webdav，其他方法的原理就是：`避开zotero.sqlite-journal`。
1. 我的方案是直接把C:\Users\[Username]\Zotero整个数据库搬到了云（小米云，实时同步），开保留本地，要求就是：**打开Zotero时同步软件要关闭，要同步时Zotero要关闭**（zotero.sqlite-journal生成和消失时，云同步不在场）。
2. 另一种做法的Syncthing去中心化同步，得设备一直开机或者中转，嫌麻烦没弄。我认为只要不是实时同步的云，比如只能手动同步的云都可以用。

**Reply by Charagol** [reply]
补充一下我的使用场景：主要在工位的台式肝论文，另有一个笔记本可能用到，需要同步。
这种做法我认为好处也有，就是一切操作和本地的没两样，插件也好数据也好，新增文件都能直接拖，
为了避免啥时候打开zotero时忘记关云同步，我默认云同步都是关闭的，只在工位这边下班的时候，打开云更新一下就行
（更极端的，如果你有实时同步的需求，可以给zotero专门一个网盘用于下班时候打开同步，另一个一直开着实时同步）

**Reply by northword** [reply]
那也可以：

- 官方同步同步条目信息和注释笔记等
- 把 zotero 数据文件夹除 sql 文件外的目录硬连接到网盘，使网盘同步附件、styles、translators
- 把 profile 硬连接到网盘，使网盘同步插件及其配置。

把sql直接放在网盘风险还是相对大了一些，万一哪次没记得关网盘同步，数据库就有可能损坏了。

**Reply by Charagol** [reply]
直接webdev了，真的很简单只需要10分钟搞定并且0门槛，海外的可以试一下teracloud，不需要梯有20g，我先用着后面看看

**Reply by RerrentLinden** [reply]
实测用onedrive备份C:\Users[Username]\Zotero整个数据库是可行的。
如果在打开zotero之前没打开onedrive同步，确实会出现同时存在2个zotero.sqlite文件的情况。
但onedrive比较聪明，此时它会把云端的zotero.sqlite文件加上对应上传机器的名称，如zotero-GA-17.sqlite。这样就不会因为命名重复和本地文件打架。
同时运行zotero和onedrive也不会出现报错问题，因为zotero一直占用zotero.sqlite文件，onedrive会在zotero进程结束后才进行备份。
倘若习惯了把onedrive设置成开机自启，那么zotero使用的全过程还是比较无感的，完全不会遇到报错的情况。
简单点还是推荐注册个支持webdav的云盘，实在不想注册喜欢折腾还可以在自己的vps上或本地nas部署alist挂载onedrive，再通过alist实现webdav

**Reply by duoxiangqinzuo** [reply]
是不是只是zotero.sqlite-journal和zotero.sqlite这两个文件比较敏感？如果是的话，是不是把网盘设置成定时自启，比如在夜间不工作时开始同步，就安全了？

**Reply by ZhouYu1014** [reply]
百度云是产生一个个的冲突文件，这个应该怎么处理

**Comment by FeiLZX3Km** [comment]
你好，Zotero同步页面没有提供Webdav同步方式？
编辑——设置——同步——文件同步，使用 Zotero 改成 WebDAV 用这个方法找不到。

**Reply by northword** [reply]
![Screenshot_2024-04-14-19-55-46-814_com.microsoft.emmx.jpg](https://github.com/zotero-chinese/wiki/assets/44738481/b764745e-a3b2-449c-ae3d-7e6e6b495ea2)

**Reply by zhoudujin** [reply]
一模一样，我也没有webdav选项

**Reply by northword** [reply]
需要登录 zotero 账号。

**Comment by jin8111** [comment]
数据同步才是刚需啊。。文件同步太多方法可以实现了

**Comment by FranticEr** [comment]
我这边遇到的问题是有些pdf/文件在数据存储位置中存在，但是在已连接附件的根目录中不存在，导致同步不了，这是什么问题？

**Reply by northword** [reply]
你使用的什么同步方式？正确配置 WebDAV 的话，在数据存储位置的都会同步。

如果是直接同步 `已连接附件的根目录` 这个文件夹的话，需要附件为链接的附件，地址需要在这个目录。

**Reply by FranticEr** [reply]
使用的是坚果云，有的文件被同步到已连接的根目录中，有的没有（就是数据存储位置的目录中有这个pdf，而已连接附件的根目录中该文件夹只有条目没有pdf）

**Reply by FranticEr** [reply]
我认为不是同步方式的问题，而是正如遇到的问题表现的那样，是zotero本身没有将两个文件目录中（数据存储位置的目录和已连接附件的根目录）文件进行同步，坚果云的功能只是将本地的已连接附件的根目录同步到云端而已

**Comment by zixuan-go** [comment]
请问在win端中以Zotero PDF形式导出了文献库，只保留storage文件夹，如何能在新的zotero软件上导入？

**Reply by northword** [reply]
只有 pdf 的话只能按 根据附件导入条目 的方式。

建议重新备份 Zotero: https://zotero-chinese.com/user-guide/backup

**Comment by quanhui1** [comment]
您好，请问我同步后登录zotero官网后查看storage还是占用满了，请问这是正常的吗

**Comment by zhoudujin** [comment]
为啥我的zotero同步里面没有文件同步和webdav选项呢？

**Reply by northword** [reply]
登录 zotero 账号没有？

**Comment by HappyWuJunjie** [comment]
我有一个问题，123云盘我的windows上能同步，但是在ipad上就会提醒iOS: Can't connect to WebDAV server，或者就是说ios限制，只能https格式，但我的链接就是https，不知道为什么ipad不行，有么有大佬帮忙解决。debug：D250338960

**Comment by guagualiu** [comment]
请问文件同步到坚果云之后为什么是 prop或者 zip 的格式，名字也是数字和字母的组合方式。有什么办法可以更改文件格式以及命名吗？

**Reply by northword** [reply]
这是正常的，无需更改。

**Reply by guagualiu** [reply]
嗯好的！

**Comment by guagualiu** [comment]
请问在完成最后一步之后，是否需要在PC端的坚果云app中点击同步到本地？还是每次直接网页访问就可以

**Reply by northword** [reply]
不需要坚果云 PC 端。不需要网页访问坚果云，Zotero 可以自动下载文件。

**Reply by guagualiu** [reply]
谢谢！刚在另一个电脑登录 zotero，明白你说的了！

**Comment by yiye2005** [comment]
请问为什么坚果云中的zip文件打开之后只有网页链接呢？坚果云同步的文件中没有我导入Zotero的PDF文件，求告知

**Comment by polotreetion** [comment]
有人使用zotero7+坚果云webdav同步部分数据后出现这个报错吗，无法解决

Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFileInputStream.init]

**Comment by biancazhan** [comment]
你好，我想问下坚果云的网站授权对于想同步数据的电脑和ipad需要使用一个还是两个呢？我试了用同一个密码和一个新授权的密码都不行，说webdav server did not accept the username and password you entered.请问要怎么解决呢？

**Reply by northword** [reply]
同一个与否不影响，确认一下用户名、地址、密码是正确的？

**Reply by biancazhan** [reply]
谢谢！问题解决了，不是我在ipad上的信息输错了，是我在注册坚果云时的邮箱多了个数字，难怪一直折腾ipad的输入信息都没找出问题，笑死……

**Comment by biancazhan** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，您的邮件我已收到，谢谢。                                         詹李源

**Comment by VirtuousContract** [comment]
你好。请问我用webdav同步完成后，zotero官网的存储空间还是占满了，这会不会影响条目信息和笔记数据的同步？有办法把zotero官方存储中的附件删除以把空间腾出来吗？

**Reply by threegold116** [reply]
在zotero的网站 https://www.zotero.org/settings/storage 这里有个“Purge Storage in My Library”选项可以清空

**Comment by 42-0p6180339887** [comment]
小白想请问一个关于插件同步的问题。有台式机和笔记本两台机器，都已经登zotero账号并设置好坚果云的webdav同步，现在文献目录和文献附件能够同步，但是安装的插件和附件上做的高亮下划线等笔记不能同步。请问是哪里的问题，我该如何解决。谢谢大佬们

**Comment by hongtaoxuu** [comment]
阿里云盘好像支持了webDAV

**Comment by gouweicao20** [comment]
你好，我使用的同步方式是坚果云，但是由于我的文献比较多，第一个月的流量已经用完了，但是没有全部同步完成。第二个月流量到了以后，继续同步是在第一个没有同步完成的地方继续同步剩余文献，还是重新同步覆盖以前同步的文献。

**Reply by northword** [reply]
继续增量同步，不会覆盖。

**Reply by gouweicao20** [reply]
十分感谢

**Comment by gouweicao20** [comment]
你好，我在使用坚果云同步的时候，发现同步的内容有缺失，我需要同步的内容也即zotero数据库storage文件夹下有564个项目，但是第一个同步的时候流量使用完了，在第二个月同步继续同步的时候发现同步了518个，造成了同步数据丢失。这种应该怎么办？

**Comment by ch1mai** [comment]
WebDAV 同步选择太少了，基本就坚果云，但真的不是很能信任国内网盘。
其实我感觉 Dropbox Backup 可能也是个选择，因为文件是复制而不是替换的。

**Comment by Luffy-19961105** [comment]
请问关于zotero的文件同步，如果使用已经超出了300M是否有影响？会对哪些使用功能有影响？是否一定要使用WebDAV进行文件的同步才会比较好？请好心人帮忙解答一下，感谢。。。

**Comment by Molly-Xia** [comment]
两个小问题，在设置pdf等附件进行webdav设置后，1、数据同步-同步全文内容这个需要勾选吗？2、文件同步-使用 Zotero 云存储同步群组文献库中的附件需要勾选吗？

**Comment by YajingDi** [comment]
您好，我按照步骤操作后在zotero同步“验证服务器”时报错：“发生未知错误。请检查文件同步设置或联系您的 WebDAV 服务器管理员。HTTP 503”。请问您知道怎么解决吗

**Reply by YajingDi** [reply]
zotero报告ID 1815198580

**Comment by wangfh5** [comment]
我在思考有没有办法“既要还要”，也就是主要使用zotero+webdav同步附件，但同时也把附件**单向镜像**到一个类似于attanger管理的文件夹里面，再另外使用网盘进行云同步。这样既保证了主要平台和移动端的使用（PDF，批注以及`zotero://`格式的（深度）链接），又可以在次要平台（如尚未有zotero的鸿蒙端）以及web端上查看PDF。

当然，在次要平台上对PDF进行的修改是无法保存的。会被单向镜像覆盖掉。

为了实现这个功能，需要有一个类似于attanger的插件，但是它并不会将PDF附件替换为链接，只是管理zotero文献库和目标备份文件夹的匹配程度。

**Reply by wangfh5** [reply]
https://github.com/wileyyugioh/zotmoov 支持 copy 模式

**Reply by wangfh5** [reply]
最近, 这种PDF附件的单向镜像管理, 又萌生出了另一个应用方向, 就是应用于AI平台如notebookLM的粗读之中. 

将zotero中管理的文献添加到AI app中是略显繁琐的. 需要右键附件-在Finder中打开-复制文件-在聊天框中粘贴. 如果需要在notebookLM和ima这样的平台上进行多文献阅读, 更是麻烦, 需要逐个附件找到路径, 然后拖入文件夹之中. 
(希望能做一个zotero插件, 右键直接复制PDF附件本身, 可以省去打开Finder的操作)

将PDF单向镜像同步到另一个文件夹, 全部陈列在这个文件夹内, 当然也可以按照主题来进行分级. 这样的结果是, 在将PDF附件导入到AI app上就会很方便, 可以多选, 有些平台还可以整个文件夹上传. 
还可以进一步将这个文件夹, 使用Google Drive来进行同步, 那么你甚至可以在手机app上对这些PDF附件, 进行随时随地的对话学习. 

---

虽然如此, 但我觉得将PDF管理与PDF对话打通, 还是需要呼唤一个全新的软件. zotero或许会被下一代文献管理软件革命. 

目前, zotero是最好的PDF管理软件, gemini和notebookLM是最好的ChatPDF平台, 两者之间具有鸿沟. 而许多在做所谓"AI文献管理"的平台, 如玻尔科学空间、小绿鲸, 在管理上远逊于zotero, 在AI对话上远逊于Google系软件, 可谓"不堪大用".

**Comment by sixi0828** [comment]
插件一直打不开zotero，手动配置成功了

**Comment by cwlyj001** [comment]
怎么会同时同步到zotero云和webdav坚果云？
在安装了zetero后，默认是同步附件到zetero云。后来改为同步附件到webdav，
也配置好了协议参数，但附件还是会同步到zetero云上。是zotero的问题还是我的问题？

**Comment by Iris2762** [comment]
非常有用，解决了我的问题。我换电脑，之前设置的同步信息忘掉了，不知道咋同步。看了你的教程，成功同步了。

还有一点小经验想分享，如果坚果账户信息-安全选项显示-第三方应用管理显示：用于授权第三方 WebDAV 应用访问您的云空间。请按照此步骤关联：① 下载并安装坚果客户端；② 登录后前往 “设置” → “第三方应用管理” 进行关联。

只需要按照提示下载安装，然后在“设置” → “第三方应用管理” 部份就可以看到帐户和密码。

**Comment by WjqGW2020** [comment]
同步时，WebDAV 文件同步时发生错误。请重启 Zotero ，然后尝试重新同步。

如果您频繁收到此条信息，请在Zotero首选项的同步选项卡中检查 WebDAV 服务器设置。
已经按照要求进行操作后，依然有此消息提醒，请问如何解决？
